peer_review_data/main.py

```diff
@@ -99,103 +96,116 @@ def saveAssessmentsAndComments(
-        canvasAssignment)
-    LOGGER.info(f'Saving {assignment}…')
-    assignment.save()
-
-    LOGGER.info(f'Saving submissions for {assignment}…')
-    saveSubmissions(canvasAssignment)
-
-    LOGGER.info('Saving rubric and criteria…')
-    rubric, criteria = saveRubricAndCriteria(canvasAssignmentRubric,
-                                             canvasAssignment)
-
-    LOGGER.info('Saving assessments and comments…')
-    saveAssessmentsAndComments(canvasAssignmentRubric.assessments)
+    courseId: str
+    for courseId in [c.strip() for c in config.COURSE_IDS_CSV.split(',')]:
+        canvasCourse: CanvasCourse = canvas.get_course(courseId)
+        LOGGER.info(f'Found course ({canvasCourse.id}): "{canvasCourse.name}"')
+        courseSaved = False
+        course: models.Course | None = None
+
+        canvasAssignments: List[CanvasAssignment] = \
+            canvasCourse.get_assignments()
+
+        canvasAssignment: CanvasAssignment
+        # Iterate over assignments which have peer reviews
+        for canvasAssignment in filter(
+                lambda a: a.peer_reviews, canvasAssignments):
+            LOGGER.info(
+                f'Found peer reviewed assignment ({canvasAssignment.id}): '
+                f'"{canvasAssignment.name}"')
+
+            assignmentRubricId: int = canvasAssignment.rubric_settings.get(
+                'id')
+            LOGGER.debug(f'Assignment ({canvasAssignment.id}) has '
+                         f'rubric ID ({assignmentRubricId})')
+
+            canvasAssignmentRubric: CanvasRubric = canvasCourse.get_rubric(
+                assignmentRubricId, include='assessments', style='full')
+
+            if not hasattr(canvasAssignmentRubric, 'assessments'):
+                LOGGER.info(f'Skipping assignment ({canvasAssignment.id}) in '
+                            f'course ({courseId}): '
+                            'Not configured for peer reviews ("assessments").')
+                continue
+
+            LOGGER.info(f'Assignment ({canvasAssignment.id}) '
+                        f'in course ({courseId}) is '
+                        'configured for peer reviews ("assessments")…')
+
+            if len(canvasAssignmentRubric.assessments) == 0:
+                LOGGER.info(
+                    f'Skipping assignment ({canvasAssignment.id}) '
+                    f'in course ({courseId}): '
+                    'No peer reviews ("assessments") were found.')
+                continue
+
+            LOGGER.info(f'Assignment ({canvasAssignment.id}) '
+                        f'in course ({courseId}) has '
+                        'peer reviews ("assessments")…')
+
+            if not courseSaved:
+                course = saveCourseAndUsers(canvasCourse)
+                if course:
+                    courseSaved = True
+                else:
+                    LOGGER.error(f'Course ({canvasCourse.id}) '
+                                 'and its users were not saved.')
+                    continue
```

Comment:
Are you waiting to do this because you only want to save the course if it has peer reviews? This seems fine, but would be nice to move it out of the for loop. Maybe you could somehow check whether any of the assignments have peer reviews first, then start looping? But hard to tell in advance if there's actually data without digging into it all. This is probably fine.

Reply:
Yeah, I didn't want to save the course and the user data unless we know for sure whether the course's many assignments are peer reviewed and have assessments (peer review comments).  I'll look at it again to see whether it could be done more gracefully.

Reply:
I've tried to simplify the logic somewhat.  Waiting to save the course until it's known whether there are peer review assignments with assessments still appears to be the correct thing to do.